### ISSUE TITLE
supporting new versions of pgspecial

### ISSUE THREAD

**Issue by edublancas** [ploomber/jupysql#367]
background: https://github.com/ploomber/jupysql/issues/347 Looks like pgspecial breaks if using psycopg2, since new versions need psycopg3. So we need to tell the user how to get it working.I'm thinking the workflow should go like this:

1. User tries to run a meta-command e.g. `\d`
1. We show a "missing pgspecial error" but customize it depending on which psycopg2  version they're using (because if they use psycopg2, they need a lower version of pgspecial)

I think this might impact sqlalchemy asl well. unsure if sqlalchemy has support for psycopg3.

**Comment by tl3119** [comment]
For this task, the basic idea is that we need to edit in [this file](https://github.com/ploomber/jupysql/blob/master/.github/workflows/ci-integration-db.yaml) to check whether pgspecial will break and if breaks, we need to tell the user how should they do, such as need a lower version of pgspecial. 
code likes:
`python -c "import psycopg2, sqlalchemy; print('Psycopg2 version', psycopg2.__version__); print('Sqlalchemy version', sqlalchemy.__version__)"
psycopg2_version = psycopg2.__version__.split(' ')[0]
if psycopg2_version >= '2.9':
    error_msg = 'Please upgrade psycopg2 to a version that supports psycopg3.'
else:
    error_msg = 'Please install a lower version of pgspecial that supports psycopg2. '
`
is that correct do i miss something?

**Comment by edublancas** [comment]
so this one needs a bit more investigation. 

currently, jupysql supports sqlalchemy 1.x and sqlalchemy 2.x. we also support psycopg2

my conclusion from some initial research is that with this combination of versions, we can only support old versions of pgspecial. the objective is to support newer versions of pgspecial, but to support that we might need to do some extra adjustments. so the first task is to figure out what versions of sqlalchemy and psycopg we need to support new versions of pgspecial, then implement support for it. finally, we can determine if it's feasible to support both old and new versions of pgspecial

**Comment by tl3119** [comment]
For the information I investigate, we need psycopg3 (>=3.0.10) [version ](https://libraries.io/pypi/pgspecial/2.1.0#:~:text=(%3E%3D4.1)-,psycopg,-(%3E%3D3.0.10))to support new versions of pgspecial.

**Comment by edublancas** [comment]
ok, can you see if jupysql works with psycopg3? my guess is that you'd need sqlalchemy 2 (which we support). I think let's first find out about psycopg3 and then we can continue with pgspecial

**Comment by tl3119** [comment]
I change `psycopg2-binary` in [setup.py](https://github.com/ploomber/jupysql/blob/master/setup.py#LL59C6-L59C6) to `psycopg3` and then activate the virtual environment. After that, I run the command `pip install -e .`, and all the package install successfully. Does that mean jupysql can work with psycopg3? Or do I need to do other tests to check the condition?

**Comment by edublancas** [comment]
yes, you need to connect to a database and run a few queries. you can use this as a reference: https://jupysql.ploomber.io/en/latest/integrations/postgres-connect.html just modify the connection string and ensure it uses psycopg3

**Comment by tl3119** [comment]
I am a little confused how do I test this locally to make sure it uses psycopg3?

**Comment by tl3119** [comment]
It seems it successfully installed psycopg3 after I activate the virtual environment.
<img width="979" alt="image" src="https://github.com/ploomber/jupysql/assets/71258902/a4772ecc-90bb-4b27-ba5c-fc5b6bcefaed">

**Comment by edublancas** [comment]
> you need to connect to a database and run a few queries.

**Comment by tl3119** [comment]
> > you need to connect to a database and run a few queries.

Sure! I connect to a database and run the queries, it can work!
![image](https://github.com/ploomber/jupysql/assets/71258902/d50707bc-49c8-4962-8293-89b928e536e0)
![image](https://github.com/ploomber/jupysql/assets/71258902/7ca0d72b-800c-4140-a75c-dcb29f5d37d6)
![image](https://github.com/ploomber/jupysql/assets/71258902/ba65b02f-6c36-42aa-a3d8-b09888c70802)
